gsy_framework/matching_algorithms/pay_as_bid_matching_algorithm.py

```diff
@@ -15,33 +15,78 @@ class PayAsBidMatchingAlgorithm(BaseMatchingAlgorithm):
            since the most affordable offers will be allocated for the most aggressive buyers.
     """
 
+    @staticmethod
+    def _match_one_bid_one_offer(offer: Dict, bid: Dict, already_selected_order_energy: Dict,
```

Comment:
AFAIS this name is misleading `already_selected_order_energy`, it gives the impression that it's holding track of the amount of energy that was already consumed for each order, whereas at the end of the iteration it will in fact be holding the available energy per each order

Reply:
I was also wondering whether we may end up with 2 orders with the same id and how we should handle it

Reply:
Regarding the first comment, I agree that this is misleading, and I will rename it as you suggested to `available_order_energy`.

Regarding the latter, I think we have discussed it in the past that hash collisions with uuid4() are very very rare, therefore I think that we can safely ignore them. If it would make you feel safer, I think that this will need to be handled by either the Myco or the market validator, which will essentially loop over all the orders and validate that there are no hash collisions. I find it an overkill, if you would like it though I can open a PR that deals with this.

Reply:
Yes I always knew that this hash collision is a rare situation and yet I'm always anxious about it 😃 , don't bother with it then, Thanks!